packages/calypso-codemods/transforms/replace-lodash-get.js

```diff
@@ -0,0 +1,171 @@
+/**
+ * Transform all safely recognizable instances of _.get().
+ * Conditions include:
+ * - Must be imported in the form `import { get, ... } from 'lodash'` or `import { get as foo, ... } from 'lodash'`
+ * - Must either not use a default parameter, or use a default parameter set to the `null` literal
+ * - Must use a string literal, string template, or array expression as the path
+ *
+ * If at least one of the conditions is not met, the transformation will be dropped.
+ *
+ * In files with at least one successful transformation, the import is removed if no longer used.
+ */
+
+// Transform function.
+export default function transformer( file, api ) {
+	const j = api.jscodeshift;
+	const root = j( file.source );
+
+	let getMethodName;
+	let visitedGetCalls = 0;
+	let modifiedGetCalls = 0;
+
+	function debug( text ) {
+		throw new Error( `${ file.path }: ${ text }` );
+	}
+
+	function isValidIdentifier( string ) {
+		return string.match( /^[A-Za-z]\w*$/ );
+	}
+
+	function partToLiteral( string ) {
+		const partAsNumber = Number.parseInt( string, 10 );
+		return j.literal( isNaN( partAsNumber ) ? string : partAsNumber );
+	}
+
+	const lodashImports = root.find( j.ImportDeclaration, {
+		source: {
+			type: 'Literal',
+			value: 'lodash',
+		},
+	} );
+
+	// Look for imported `get` method.
+	lodashImports.forEach( nodePath => {
+		const specifier = nodePath.value.specifiers.find( s => s.imported.name === 'get' );
+		if ( specifier ) {
+			getMethodName = specifier.local.name;
+		}
+	} );
+
+	if ( getMethodName ) {
+		root
+			.find(
+				j.CallExpression,
+				node => node.callee.type === 'Identifier' && node.callee.name === getMethodName
+			)
+			.forEach( getPath => {
+				try {
+					visitedGetCalls += 1;
+
+					const { length } = getPath.value.arguments;
+					const [ object, path, defaultValue ] = getPath.value.arguments;
+
+					const hasDefaultNull =
+						length === 3 && defaultValue.type === 'Literal' && defaultValue.value === null;
+
+					if ( length !== 2 && ! hasDefaultNull ) {
+						return;
+					}
+
+					let pathElements;
+
+					if ( path.type === 'Literal' && typeof path.value === 'string' ) {
+						// String-based path
+						pathElements = path.value
+							.split( /[.[]/ )
+							.filter( el => el !== '' )
+							.map( el => {
+								const part = el.replace( ']', '' );
+								if ( isValidIdentifier( part ) ) {
```

Comment:
Seems like we use that piece for array paths too, should we define it at a single place and reuse it in all locations?

Reply:
Do you mean extracting the `if/else` logic to a method too? I can do that, sure!

Reply:
Yeah, because it's used when handling arrays below, too. Thanks!

Reply:
Done!